src/Sylius/Component/Core/Promotion/Applicator/AdditionalFiltersApplicator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of Sylius corporate website.
+ *
+ * (c) Sylius <sylius+sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Promotion\Applicator;
+
+use Sylius\Component\Core\Promotion\Filter\FilterInterface;
+use Webmozart\Assert\Assert;
+
+final class AdditionalFiltersApplicator implements AdditionalFiltersApplicatorInterface
```

Comment:
it's a great suggestion, and it makes the code simpler. I no longer need my `AdditionalFiltersApplicatorInterface` and the associated service.

However, I have to use the same configuration for all filters, and currently there are 2 types of configuration:

``php
        $filteredItems = $this->priceRangeFilter->filter(
            $subject->getItems()->toArray(),
            array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
        );
        $filteredItems = $this->taxonFilter->filter($filteredItems, $configuration[$channelCode]);
```

I think I can pass the most complete array for all filters, namely: `array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode])`. WDYT?

Reply:
Yep, don't see what can be done about this, so

```php
$this->compositeFilter->filter(
    $subject->getItems()->toArray(),
    array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode])
); 
```

and each filter will have access to channel instance instead of just priceRangeFilter. No harm in this, especially that each `$configuration` has `$channelCode`

Another point, action sends `$configuration[$channelCode]` into filter instead of full `$configuration` and leave filter to decide how to access configuration, but it's unsolvable without BC as far as I can see.